features/dd-sdk-android-trace/src/main/java/com/datadog/opentracing/PendingTrace.java

```diff
@@ -127,21 +127,23 @@ public void dropSpan(final DDSpan span) {
   }
 
   public void addSpan(final DDSpan span) {
-    if (span.getDurationNano() == 0) {
-      return;
-    }
-    if (traceId == null || span.context() == null) {
-      return;
-    }
-    if (!traceId.equals(span.getTraceId())) {
-      return;
-    }
+    synchronized(this) {
+      if (span.getDurationNano() == 0) {
+        return;
+      }
+      if (traceId == null || span.context() == null) {
```

Comment:
I'm assuming that the reason all fail fast conditionals are separated is so that we can figure out exactly which one caused the problem in a stacktrace. If so, perhaps we should split this one also?